### PR TITLE
Select: change arguments of the onFilter callback

### LINKED ISSUE #5625: Incorrect type for `onFilter` prop of Select component - can break an app when Select children are updated
**Describe the issue. What is the expected and unexpected behavior?**

The TypeScript type of Select's `onFilter` prop is `(e: React.ChangeEvent<HTMLInputElement>) => React.ReactElement[]`, implying that the event passed to it will always be defined/truthy. It appears that after https://github.com/patternfly/patternfly-react/pull/5375, `onFilter` gets called with a `null` value for its event argument when the component's children change. See https://github.com/patternfly/patternfly-react/blame/fe2da75b77e4cbf92ef11b71e87aef998fa8dd46/packages/react-core/src/components/Select/Select.tsx#L209

When I upgraded PF, this caused an error in my code which passes a handler to `onFilter` expecting the event to always be defined and non-null. The example in our docs doesn't hit it because it has a try-catch block when it references the event.

**Please provide the steps to reproduce. Feel free to link CodeSandbox or another tool.**

See this CodeSandbox: https://codesandbox.io/s/laughing-forest-o01j4?file=/index.js
It is based on the one from [the Custom Filtering example](https://www.patternfly.org/v4/components/select#custom-filtering) in the docs, with the try-catch in `onFilter` removed and a new state property introduced that forces the children to change after the component mounts.

**Is this a bug or enhancement? If this issue is a bug, is this issue blocking you or is there a work-around?**

This is a bug. The workaround is to use a try-catch or otherwise defensively code around the possibility of the event being null, but TypeScript could catch this for consumers if the `onFilter` prop type was changed to `(e: React.ChangeEvent<HTMLInputElement> | null) => React.ReactElement[]`.

**What is your product and what release version are you targeting?**

Migration Toolkit for Virtualization. No particular target release.

### PR DESCRIPTION
**Pull request by KKoukiou**
When the children of the select are updated while the component is
mounted make sure to apply the filter as well.

Add a demo to demostrate the new functionality.

Fixes #5625

### CODE REVIEW

File: packages/react-core/src/components/Select/examples/Select.md [hunk 1]
```diff
@@ -589,8 +589,7 @@ class FilteringCheckboxSelectInput extends React.Component {
       }
     };
 
-    this.onFilter = evt => {
```
Comment (mturley) [inline packages/react-core/src/components/Select/examples/Select.md:592]: These `this.onFilter` methods in the examples also need to be changed to have `textInput` as the second arg now I think

**Comment by mturley** [comment]
Thanks for fixing this so quickly @KKoukiou !

File: packages/react-core/src/components/Select/examples/Select.md (hunk 1 above)
Comment (KKoukiou) [reply packages/react-core/src/components/Select/examples/Select.md:592]: Good catch, thanks.

**Review by mturley — APPROVED** [review]
LGTM!

**Review by tlabaj — APPROVED** [review]
looks good!

**Review by kmcfaul — APPROVED** [review]

**Pull request merged**